benchmarks/experiment_runner.py

```diff
@@ -90,6 +91,8 @@ def run(self):
           command = ([sys.executable] + sys.argv +
                      [f"--experiment-config={experiment_config_str}"] +
                      [f"--model-config={model_config_str}"])
+          if not self._args.timestamp:
+            command += [f"--timestamp={self.timestamp}"]
```

Comment:
This function (`run`) is probably the right place to generate the timestamp.

Reply:
(Answering your other comments for this file here.)
This is a bit subtle so maybe I should have added a comment.
We could always use args.timestamp and set time.time() as its default value. Let's call that approach A.
But then in this hunk we might pass --timestamp twice to the child process; think for instance when the original invocation had --timestamp passed to it. It's no big deal (it would be --timestamp foo --timestamp foo), but I thought it would be cleaner not to do that.
Alternatives would be:
Approach B: Look for --timestamp in sys.argv. I really wouldn't bother doing this.
Approach C: The approach you see here. That is, have a --timestamp flag that defaults to None so that we know we have to pass --timestamp to the subprocess. Note that the member variable (or equivalent) is necessary in this case, since from save_results we need to get the timestamp and --timestamp cannot be set to a default [unless we're OK possibly passing --timestamp twice to the child].

I'm OK with either A or B. Let me know if you have a preference.

Reply:
Got it. I had assumed it was propagated through the metrics. Now that you outlined all of these options, I actually like approach A. Happy with either option (except for B maybe)

Reply:
OK, will do approach A, which, BTW, is what I had implemented originally x_x